src/service/ApiContainer.ts

```diff
@@ -74,7 +74,7 @@ export class ApiContainer {
   // todo this needs a home and a test
   private wrap(controller: Function): Middleware {
     return async (ctx: Context, next: Next) => {
-      const input = ctx.request.body || ctx.request.query;
+      const input = ctx.params.id || ctx.request.body || ctx.request.query;
```

Comment:
please change it to `const input = ctx.params || ...` so that all the params get passed.